### ISSUE TITLE
Fix Scale9Sprite to work with Global Z Order

### ISSUE THREAD

**Issue by tmsoft** [cocos2d/cocos2d-x#10435]
Scale9Sprite does not support usage of Global Z Order.  This is because the visual elements are stored as protected children outside of the standard children array.  Overriding the setGlobalZOrder method with the following code fixes this issue:

```
void Scale9Sprite::setGlobalZOrder(float globalZOrder)
{
    Node::setGlobalZOrder(globalZOrder);
    // apply to protected children too
    for( auto &child: _protectedChildren)
        child->setGlobalZOrder(globalZOrder);
}
```

**Comment by Dimon4eg** [comment]
Hi,
It's still reproduce in cocos 3.8.1. Issue duplicate - #13943
Fix is simple, why it's not yet fixed?

**Comment by zilongshanren** [comment]
@Dimon4eg Since it is duplicated, I will close this one and let's try to fix the #13943
